src/components/SlateEditor/plugins/embed/EditImage.jsx

```diff
@@ -1,73 +1,196 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+import FocusTrapReact from 'focus-trap-react';
+import Types from 'slate-prop-types';
 import { injectT } from '@ndla/i18n';
-import { css } from 'react-emotion';
-import SlateInputField from './SlateInputField';
+import { spacing, colors, animations, shadows } from '@ndla/core';
+import Button from '@ndla/button';
+import { FormInput, StyledButtonWrapper } from '@ndla/forms';
+import { isEmpty } from '../../../validators';
 import ImageEditor from '../../../../containers/ImageEditor/ImageEditor';
 import { Portal } from '../../../Portal';
-import { EmbedShape } from '../../../../shapes';
+import { EmbedShape, EditorShape } from '../../../../shapes';
+import { getSchemaEmbed } from '../../editorSchema';
+
+const StyledInputWrapper = styled.div`
+  background: ${colors.brand.greyLightest};
+  padding: ${spacing.normal};
+  position: relative;
+  z-index: 20;
+`;
+
+const Background = styled.div`
+  position: fixed;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  background: rgba(0, 0, 0, 0.3);
+  z-index: 10;
+  ${animations.fadeIn()}
+`;
+
+const editorContentCSS = css`
+  box-shadow: ${shadows.levitate1};
+`;
 
 const imageEditorWrapperStyle = css`
   background-color: white;
 `;
 
 class EditImage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alt: props.embed.alt,
+      caption: props.embed.caption,
+      imageUpdates: undefined,
+      madeChanges: false,
+    };
+    this.onUpdatedImageSettings = this.onUpdatedImageSettings.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.onAbort = this.onAbort.bind(this);
+  }
+
   componentDidMount() {
     const { placeholderEl, embedEl } = this;
```

Comment:
Kan vi forenkle denne funksjonen litt? I stedet for at vi forskyver innholdet bak, så syns jeg det er mer logisk å legge modalen oppå, og ikke endre layouten til editoren under. Altså ikke endre style på placeholderEl, og bare beholde <StyledButtonFigure i DOMen.

Reply:
I utgangspunktet enig, men hvis vi forenkler og beholder bilde i bakgrunnen får vi fra noen bildestørrelser dobbelt opp ved at bilde bak blir synlig da medium sizes feks er -25% margin på. Så foreslår vi beholder den slik den er?